### ISSUE TITLE
cleanUrls plugins for 6.x does not appear to work

### ISSUE THREAD

**Issue by bennage** [docpad-archive/extras#9]
The first problem I encountered was with:
`database.findAll(outPath: $endsWith: '.html')`

https://github.com/bevry/docpad-extras/blob/master/plugins/cleanurls/src/cleanurls.plugin.coffee#L16

The query does not return any results. When I examine the underlying collection, the outPath property was undefined. I queried using extensionRendered = html and that returned results.

Also, the forEach was directly off the query. It looks like it should be off the models property.

I'd like to submit a pull request, but I am unsure about the following:
1. how do I run the unit test for the plugin?
2. is the unit test for the plugin correct? I would have expected the [requested url](https://github.com/bevry/docpad-extras/blob/master/plugins/cleanurls/src/cleanurls.tester.coffee#L23) to omit the .html.
3. I'm not sure what to do with the query results. I read through some other issues and I'm guessing it needs to be added to document.urls? Though I'm not sure how that maps to the existing code.

**Comment by balupton** [comment]
In answer to your questions;
1. clone docpad, cd into docpad, `npm install`, `npm link`. Clone docpad-extras, cd into plugin, run `npm install`, `npm link docpad`, then `npm test` to test the plugin - will document this in a wiki page (feel free to do it for me!)
2. I would have to seriously doubt they are correct, as the unit tests pass!
3. As of DocPad v6, the clean url will be the default url. Perhaps instead of the two line add then set, we should introduce a new `setUrl` that does both.

The problem seems to be with what you described, the findAll returning nothing. Perhaps the event fires too early, before they have been contextualized (out path generated). We should probably change the query to be like this instead to use live collections (so it always stays up to date, no matter where we are) and change the event to something like `docpadReady` or something (doesn't exist yet).

```
database.createLiveChildCollection().setQuery('cleanurls', outPath: $endsWith: '.html').on 'add', (document) ->
```

As this is a bug, getting the fix out asap is most important. Will have a look at it in about 8 hours (my evening sydney time), though if you can fix it sooner that'll be grand. Otherwise if this were just a new a feature or whatever, I'd happily wait and take the time to help you as much as possible with submitting a pull request and contributing to the codebase yourself :)

**Comment by balupton** [comment]
Fixed with plugin version 2.0.2, requires docpad 6.0.8

**Comment by bennage** [comment]
Thanks! Sorry that I didn't get to it quicker myself. :-)
I'm currently working on Windows and apparently `npm link` isn't supported. Ugh. I got a little distracted on that.
